stablehlo/transforms/Passes.td

```diff
@@ -145,6 +145,21 @@ def ShapeLegalizeToStablehloPass : Pass<"shape-legalize-to-stablehlo", "func::Fu
   let dependentDialects = ["mlir::stablehlo::StablehloDialect"];
 }
 
+def StablehloLegalizeDeprecatedOpsPass : Pass<"stablehlo-legalize-deprecated-ops", "func::FuncOp"> {
+  let summary = "Legalize deprecated ops to well-supported ops.";
+  let description = [{
+    The StableHLO v1.0 Opset Deprecations RFC (#2283) proposes to remove
+    several redundant ops. This pass helps to evaluate the impact of these op
+    removals in various compilation pipelines by legalizing them to their
+    long-term supported counterparts.
+  }];
+  let dependentDialects = ["mlir::stablehlo::StablehloDialect"];
+  let options = [
+    Option<"failOnUnusedOps", "fail-on-unused", "bool", /*default=*/"true",
+           "Fail on (mostly) unused ops that are deprecated without any fallback.">,
```

Comment:
"Unused" here refers to the fact that we expect these ops to be rarely used in the wild?

Maybe `failOnUnlegalizableOps` would be more accurate?

Reply:
I think "unused" conveys the intent. Ideally this flag being true would be a no-op, but if the op is used, that's the impact we want to discover.